packages/flutter_tools/lib/src/build_system/targets/linux.dart

```diff
@@ -157,3 +154,127 @@ class DebugBundleLinuxAssets extends Target {
     );
   }
 }
+
+/// Creates a profile bundle for the Linux desktop target.
+class ProfileBundleLinuxAssets extends Target {
+  const ProfileBundleLinuxAssets();
+
+  @override
+  String get name => 'profile_bundle_linux_assets';
+
+  @override
+  List<Target> get dependencies => const <Target>[
+    AotElfProfile(TargetPlatform.linux_x64),
+    UnpackLinux(),
+  ];
+
+  @override
+  List<Source> get inputs => const <Source>[
+    Source.pattern('{BUILD_DIR}/app.so'),
+    Source.pattern('{FLUTTER_ROOT}/packages/flutter_tools/lib/src/build_system/targets/linux.dart'),
+    Source.pattern('{PROJECT_DIR}/pubspec.yaml'),
+    ...IconTreeShaker.inputs,
+  ];
+
+  @override
+  List<Source> get outputs => const <Source>[
+    Source.pattern('{OUTPUT_DIR}/lib/libapp.so'),
+  ];
+
+  @override
+  List<String> get depfiles => const <String>[
+    'flutter_assets.d',
+  ];
+
+  @override
+  Future<void> build(Environment environment) async {
+    if (environment.defines[kBuildMode] == null) {
+      throw MissingDefineException(kBuildMode, 'profile_bundle_linux_assets');
+    }
+    final Directory assetsDirectory = environment.outputDir
```

Comment:
There's a bunch of duplicated logic across these three targets. It would be good to use inheritance and/or dependent targets to share code (see Windows, which has a mix of both, as an example; it's not necessary perfect in that I don't have a good argument for that mix vs. going entirely to inheritance or dependencies, but it does avoid repeating logic 2-3 time).